OOP-Assignment1-Charles/OOP-Assignment1-Charles/Input.cs

```diff
@@ -6,10 +6,47 @@
 
 namespace OOP_Assignment1_Charles
 {
-    public class Input
+    class Input : Validate //(INHERITANCE) the input class inheirts from the validate class so inputs can be validated when taken.
     {
-        //Handles the text input for Assessment 1
+
+
         string text = "nothing";
+        string usersChoice;
+        string[] returnInput = { "", "" };
+
+
+        public string ManualOrFile()
+        {
+            string userInput = ValidateManualOrFile("Please enter eiter 'manual', or 'file: ");
+            return (userInput);
+        }
+
+        public string[] GetInput()
+        {
+            usersChoice = ManualOrFile();
+
+            if (usersChoice == "manual")
+            {
+                text = manualTextInput();
+
+                returnInput[0] = text;
+                returnInput[1] = usersChoice; // usersChoice of the two options is stored to be passed to the Analyse class so it can work out if it needs to calcualte max length of values for option two.
+
+            }
+
+            else if (usersChoice == "file")
+            {
+                text = fileTextInput();
+
+                returnInput[0] = text;
+                returnInput[1] = usersChoice;
```

Comment:
Inefficient, using two ways to return input, may be easier if you made this a normal string method and use a string variable rather than using a string array and just storing the input into one variable.